dist/gcc-compatible/EverCrypt_Hash.h

```diff
@@ -47,236 +47,47 @@ extern "C" {
-  uint8_t *dst,
-  uint8_t *input,
-  uint32_t len
-);
-
-uint32_t EverCrypt_Hash_Incremental_hash_len(Spec_Hash_Definitions_hash_alg a);
-
-uint32_t EverCrypt_Hash_Incremental_block_len(Spec_Hash_Definitions_hash_alg a);
-
-typedef struct Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s_____s
+typedef struct EverCrypt_Hash_Incremental_hash_state_s
 {
   EverCrypt_Hash_state_s *block_state;
   uint8_t *buf;
   uint64_t total_len;
 }
-Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____;
+EverCrypt_Hash_Incremental_hash_state;
 
-Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____
+EverCrypt_Hash_Incremental_hash_state
 *EverCrypt_Hash_Incremental_create_in(Spec_Hash_Definitions_hash_alg a);
 
-void
-EverCrypt_Hash_Incremental_init(Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____ *s);
+void EverCrypt_Hash_Incremental_init(EverCrypt_Hash_Incremental_hash_state *s);
 
 EverCrypt_Error_error_code
 EverCrypt_Hash_Incremental_update(
-  Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____ *s,
+  EverCrypt_Hash_Incremental_hash_state *s,
   uint8_t *data,
   uint32_t len
 );
 
-void
-EverCrypt_Hash_Incremental_finish_md5(
-  Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____ *p,
-  uint8_t *dst
-);
-
-void
-EverCrypt_Hash_Incremental_finish_sha1(
-  Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____ *p,
-  uint8_t *dst
-);
-
-void
-EverCrypt_Hash_Incremental_finish_sha224(
-  Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____ *p,
-  uint8_t *dst
-);
-
-void
-EverCrypt_Hash_Incremental_finish_sha256(
-  Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____ *p,
-  uint8_t *dst
-);
-
-void
-EverCrypt_Hash_Incremental_finish_sha3_256(
-  Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____ *p,
-  uint8_t *dst
-);
+Spec_Hash_Definitions_hash_alg
+EverCrypt_Hash_Incremental_alg_of_state(EverCrypt_Hash_Incremental_hash_state *s);
 
-void
-EverCrypt_Hash_Incremental_finish_sha384(
-  Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____ *p,
-  uint8_t *dst
-);
+void EverCrypt_Hash_Incremental_finish(EverCrypt_Hash_Incremental_hash_state *s, uint8_t *dst);
 
-void
-EverCrypt_Hash_Incremental_finish_sha512(
-  Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____ *p,
-  uint8_t *dst
-);
+void EverCrypt_Hash_Incremental_free(EverCrypt_Hash_Incremental_hash_state *s);
 
-void
-EverCrypt_Hash_Incremental_finish_blake2s(
-  Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____ *p,
-  uint8_t *dst
-);
+void EverCrypt_Hash_Incremental_hash_256(uint8_t *input, uint32_t input_len, uint8_t *dst);
 
-void
-EverCrypt_Hash_Incremental_finish_blake2b(
-  Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____ *p,
-  uint8_t *dst
-);
-
-Spec_Hash_Definitions_hash_alg
-EverCrypt_Hash_Incremental_alg_of_state(
-  Hacl_Streaming_Functor_state_s___EverCrypt_Hash_state_s____ *s
-);
+void EverCrypt_Hash_Incremental_hash_224(uint8_t *input, uint32_t input_len, uint8_t *dst);
```

Comment:
is there a difference between `EverCrypt_Hash_Incremental_hash_224` and `EverCrypt_Hash_Incremental_hash`? if so, can we unify the order of arguments and add documentation?